internal/api/middleware.go

```diff
@@ -87,35 +87,39 @@ func (a *API) limitEmailOrPhoneSentHandler() middlewareHandler {
 	return func(w http.ResponseWriter, req *http.Request) (context.Context, error) {
 		c := req.Context()
 		config := a.config
-		if (config.External.Email.Enabled && !config.Mailer.Autoconfirm) || (config.External.Phone.Enabled) {
-			if req.Method == "PUT" || req.Method == "POST" {
-				bodyBytes, err := getBodyBytes(req)
-				if err != nil {
-					return c, internalServerError("Error invalid request body").WithInternalError(err)
-				}
 
-				var requestBody struct {
-					Email string `json:"email"`
-					Phone string `json:"phone"`
-				}
+		if req.Method == "PUT" || req.Method == "POST" {
```

Comment:
```suggestion
		if (config.External.Email.Enabled && !config.Mailer.Autoconfirm) || (config.External.Phone.Enabled) {
```

Actually good to keep the original condition as it prevents double-parsing of the body. A tiny bit more efficient.